src/identities/mnemonic.ts

```diff
@@ -355,8 +355,8 @@ export default class Mnemonic extends Identity implements IdentityInterface {
       const restoredAddresses = await this.restoreAddresses();
       restoredAddresses.forEach(addr => this.persistAddressToCache(addr));
       return true;
```

Comment:
I this we should not return a boolean at all at this point. Just do the side effect in the positive case.